### PR TITLE
Colton adding moves

### PR DESCRIPTION
**Pull request by csomes**
Added moves for my anatures

### CODE REVIEW

**Comment by csomes** [comment]
Moveset’s:

Scorta:

Ember (Cinder) - 40 power, 10% chance of burn fire  
Tackle (Tackle)- 20 power normal
Slash (Smash) - 25 power normal physical
Flamethrower (FireTorture) - 90 power 10% chance of burn fire
smokescreen (SmogWave)- lower target acc by one stage


Smoldren:

Ember (Cinder) - 40 power, 10% chance of burn fire  
Tackle (Tackle)- 20 power normal
Slash (Smash) - 25 power normal physical
Flame (FireTorture) - 90 power 10% chance of burn fire
smokescreen (SmogWave)- lower target acc by one stage

Firespin (FireCell) - 100 power, burns if it hits, fire
Scary Face (Scary Face)- lower defense by 2 stages

Scaldris:

Ember (Cinder) - 40 power, 10% chance of burn fire  
Tackle (Tackle)- 20 power normal
Slash (Smash) - 25 power normal physical
Flamethrower (FireTorcher) - 90 power =20% chance of burn fire
smokescreen (SmogWave)- lower target acc by one stage

Firespin (Fire-a-cell) - 100 power, burns if it hits, fire
Scary Face (Scary Face)- lower defense by 2 stages

Flare blitz  (Flame Bout)- 120 power, 30% chance of burn


Transgendeero/a:

Tackle (Tackle) - 20 power normal
Leer (Leen) - lower target defense by 1 stage normal -DONE
Stomp (Clop) - 65 power normal
Stag horn  (Antler shot)- 75 power, lower target defense 1 stage, raise user attack 1 stage, normal

Clouder:

tackle
Thunder Shock  (Electro Sonic)- 40 power 10% paralysis, electric
Sonic boom (Stormy Breeze)- 20hp damage, normal
screech (Holler)- lower target attack by 2 stages

Cloutrain:

tackle
Thunder Shock  (Electro Sonic)- 40 power 10% paralysis, electric
Sonic boom (Stormy Breeze)- 20hp damage, normal
screech (Holler)- lower target attack by 2 stages

Flash cannon (Light Missle)- 80power, lower target special defense one stage
Thunder (Lightning)- 110 power, 30% chance of paralysis, electric

Cloudoom:

tackle
Thunder Shock  (Electro Sonic)- 40 power 10% paralysis, electric
Sonic boom (Stormy Breeze)- 20hp damage, normal
screech (Holler)- lower target attack by 2 stages

Flash cannon (Light Missle)- 80power, lower target special defense one stage
Thunder (Lightning)- 110 power, 30% chance of paralysis, electric

Zap Cannon (Electrocution) - 120 power, paralyzes target, target defense decrease one stage, electric

Belvidia:

Magical spice (Magical Spice) - 70 power, Dark 
Ancient power (Forgotten awakening)- 60 power, raises attack and defense by 1 stage, rock
Healing Wings (Healing Winds)- heals nature 1/2 of current HP, normal
Tackle

File: Anature/src/application/enums/MoveIds.java [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 public enum MoveIds
 {
-	NullMove, Tackle, Grumble, Flamethrower, Double_Punch, Skip_Turn, Pocket_Sand, Flail,
+	NullMove, Tackle, Grumble, Flamethrower, Double_Punch, Skip_Turn, Pocket_Sand, Flail, Cinder, Smash, Smog_Wave,
+	Fire_Cell, Fire_Torture ,Scary_Face, Flame_Bout, Electro_Sonic, Stormy_Breeze, Holler, Light_Missle, Lightning, Electrocution, 
+	Magical_Spice, Forgotten_Awakening, Healing_Winds, Clop, Antler_Shot, Leen
```
Comment (kenny2892) [inline Anature/src/application/enums/MoveIds.java]: Could you organize the moves by type? Makes it easier to identify what each move is.

File: Anature/src/application/pools/MovePool.java [hunk 2]
```diff
@@ -107,6 +125,186 @@ private static void generateMoves()
...
+						.doesDamage(false)
+						.isPhysicalAttack(false)
+						.withTotalMovePoints(15)
+						.withMovePower(0)
+						.withAccuracy(100)
+						.create());
+		
+		mMoves.put(MoveIds.Flame_Bout,
+				new MoveBuilder<FlameBout>(MoveIds.Flame_Bout).withName("Flame Bout")
+						.withType(Type.Normal)
```
Comment (kenny2892) [inline Anature/src/application/pools/MovePool.java]: _Flame_ Bout is a normal type? Is that correct?

File: Anature/src/application/pools/MovePool.java [hunk 3]
```diff
@@ -107,6 +125,186 @@ private static void generateMoves()
...
+						.isPhysicalAttack(true)
+						.withTotalMovePoints(20)
+						.withMovePower(90)
+						.withAccuracy(80)
+						.create());
+		
+		mMoves.put(MoveIds.Smog_Wave,
+				new MoveBuilder<SmogWave>(MoveIds.Smog_Wave).withName("Smog Wave")
+						.withType(Type.Normal)
+						.doesDamage(false)
```
Comment (kenny2892) [inline Anature/src/application/pools/MovePool.java]: Smog Wave is a normal type? Is that correct, cause it sounds like a poison type move.

**Review by kenny2892 — COMMENTED** [review]
Everything looks good, the only thing I think you could change are some of the move classes. Some could be swapped out to use the abstract classes "JustDamage" and "DamageAndStatus" instead, that way we can keep the file count down. You may also want to add in the movesets you created for your anatures to the anatureSpeciesDatabase.

Moves that could be abstracted:

- Cinder

- Clop

- Electro Sonic

- Fire Cell

- Fire Torture

- Flame Bout

- Lightning

- Magical Spice

- Smash

- Stormy Breeze

Note: there are 2 small logic bugs in those classes at the time of writing this. I made a pr to fix them. It should be fine to use them now though.

**Review by kenny2892 — APPROVED** [review]
Everything looks good!

**Pull request merged**